### ISSUE TITLE
Broken dependabot updates for NuGet in GitHub Enterprise Server 3.12.4

### ISSUE THREAD

**Issue by martincostello** [dependabot/dependabot-core#9887]
Back in February I made the below comment regarding my concern that various bugs with support for NuGet updates would make their way into GitHub Enterprise Server: https://github.com/dependabot/dependabot-core/issues/8483#issuecomment-1948305876

This past weekend we upgraded our GitHub Enterprise Server instance to 3.12.4, the latest release that was shipped on May 20th, and found that we are encountering many of the issues with NuGet support in GitHub.com that have been fixed over the last 6 months since the problems started in November 2023.

For example, the issue where dependabot claims to update multiple packages but only updates one:

![image](https://github.com/dependabot/dependabot-core/assets/1439341/d9f09cbc-e42b-47c3-b89e-74cc8439539e)

![image](https://github.com/dependabot/dependabot-core/assets/1439341/b948a23a-5746-4184-81fc-fdcbfaae2e0f)

It is disappointing that the functionality was ingested into GHES when it clearly had quality issues in the first place that were known of since November 2023, but that the fixes haven't been applied 6 months later is doubly disappointing.

When are the various fixes already made for GitHub.com for NuGet support going to be fixed in GitHub Enterprise Server?

**Comment by watercable76** [comment]
@martincostello Is this similar to the problems you saw in issue #8576?

**Comment by martincostello** [comment]
Pretty much - as far as I can tell, the majority of the issues we are seeing are those which have been subsequently fixed in GitHub.com. GHES just seems to be using a version that misses a large number of these fixes.

**Comment by martincostello** [comment]
For additional context:

- We do not have GitHub Connect enabled in our GHES instance, so all actions are served from the appliance itself
- The version of the github/dependabot-action in GHES 3.12.4 corresponds to https://github.com/github/dependabot-action/commit/867a767288b525c0b958f08538f5e61b9e5b3b93
- The corresponds to [ghcr.io/dependabot/dependabot-updater-nuget:v2.0.20231211155700](https://github.com/github/dependabot-action/blob/867a767288b525c0b958f08538f5e61b9e5b3b93/docker/containers.json#L15)
- The means we're missing [all these bug fixes](https://github.com/dependabot/dependabot-core/pulls?page=4&q=is%3Apr+label%3A%22L%3A+dotnet%3Anuget%22+is%3Aclosed+created%3A2023-12-10..2024-04-10++state%3Amerged) for NuGet
- Even if we used the [latest version from dotcom for the 3.12 branch](https://github.com/github/dependabot-action/blob/55f2d0af7ff63dc900e45447eb0ae172b16276ad/docker/containers.json#L15), we'd be missing fixes such as https://github.com/dependabot/dependabot-core/pull/9511.

Seems the majority of the issues would be fixed by:

- Merging https://github.com/github/dependabot-action/pull/1237:
- Updating the version of github/dependabot-action included inside GHES to a version containing the updated docker tags.

**Comment by TetianaPozniakovaClone** [comment]
Is there any updates on this? When could we expect for the fixes?

**Comment by martincostello** [comment]
Having installed GHES 3.13 onto a test server to inspect the contents (it's unfortunate that the versions of actions that ship inside GHES aren't documented somewhere or included in the separate installation metadata), we can see that GHES 3.13 includes https://github.com/github/dependabot-action/commit/55f2d0af7ff63dc900e45447eb0ae172b16276ad from April 10th.

This would include a large number of fixes for the issues we're experiencing, but still lacks the fix for https://github.com/dependabot/dependabot-core/pull/9511 as the commit is 6 days too early to include that fix.

It is also very disappointing that this issue has been open for a month, with additional reach out through GitHub Enterprise Support, and there has been zero response to this issue through either channel.

**Comment by abdulapopoola** [comment]
Hello there and sorry about the delay - the team has been stretched thin across multiple pressing needs.

Would it be possible to upgrade to GHES 3.13 and using actions sync to get to the latest? If that solves it, that might be a way to get through this.

Again, I apologize for the long turnaround times.

**Comment by martincostello** [comment]
> Would it be possible to upgrade to GHES 3.13 and using actions sync to get to the latest?

I've asked our Enterprise Admins but have no date yet.

I think the more long-term item here is to prevent such a broken version of dependabot getting into GHES in the first place, and for any issues to be addressed in a timely manner if they do.

**Comment by abdulapopoola** [comment]
@martincostello , yes you are right and yes, we are working on improving our test infra to avoid situations like this.

**Comment by abdulapopoola** [comment]
@martincostello , did this get resolved? If so, can we close this out?

**Comment by martincostello** [comment]
Yes and no.

While various bugs were fixed in the dependabot-core repo, it was months before a new version of dependabot-action including those changes was released that updated the relevant Git tags, and then GHES itself didn't actually include that update, so the net effect was that GHES was still broken for us.

We had to manually go into the appliance and get an Enterprise Admin to force upgrade the action.

GHES needs to ship with any critical fixes like this, otherwise out-of-the-box it's still broken for the end users.

**Comment by martincostello** [comment]
Also we just discovered that updating the version of GHES seems to reset the manual action upgrade.
